libs/react-helpers/src/hooks/use-environment.tsx

```diff
@@ -0,0 +1,125 @@
+import type { ReactNode } from 'react';
+import type { Networks } from '@vegaprotocol/smart-contracts-sdk';
+import { EnvironmentConfig } from '@vegaprotocol/smart-contracts-sdk';
+import { createContext, useContext } from 'react';
+
+declare global {
+  interface Window {
+    _ENV?: RawEnvironment;
+  }
+}
+
+type VegaContracts = typeof EnvironmentConfig[Networks];
+
+type EnvironmentProviderProps = {
+  definintions?: Partial<RawEnvironment>;
+  children?: ReactNode;
+};
+
+export const ENV_KEYS = [
+  'VEGA_URL',
+  'VEGA_ENV',
+  'ETHEREUM_CHAIN_ID',
+  'ETHEREUM_PROVIDER_URL',
+  'ETHERSCAN_URL',
+] as const;
+
+type EnvKey = typeof ENV_KEYS[number];
+
+type RawEnvironment = Record<EnvKey, string>;
+
+export type Environment = {
+  VEGA_URL: string;
+  VEGA_ENV: Networks;
+  ETHEREUM_CHAIN_ID: number;
+  ETHEREUM_PROVIDER_URL: string;
+  ETHERSCAN_URL: string;
+  ADDRESSES: VegaContracts;
+};
+
+const getBundledEnvironmentValue = (key: EnvKey) => {
+  switch (key) {
+    // need to have these hardcoded so on build time we can insert sensible defaults
+    case 'VEGA_URL':
+      return process.env['NX_VEGA_URL'];
+    case 'VEGA_ENV':
+      return process.env['NX_VEGA_ENV'];
+    case 'ETHEREUM_CHAIN_ID':
+      return process.env['NX_ETHEREUM_CHAIN_ID'];
+    case 'ETHEREUM_PROVIDER_URL':
+      return process.env['NX_ETHEREUM_PROVIDER_URL'];
+    case 'ETHERSCAN_URL':
+      return process.env['NX_ETHERSCAN_URL'];
+  }
+};
+
+const transformValue = (key: EnvKey, value?: string) => {
+  switch (key) {
+    case 'VEGA_ENV':
+      return value as Networks;
+    case 'ETHEREUM_CHAIN_ID':
+      return Number(value) ?? 3;
```

Comment:
chore (blocking): I don't think we want this to default, let's throw an error if it's not defined. I'd rather be explicit than accidentally have someone create a mainnet environment and be pulling testnet data

Reply:
sure, absolutely. updated so we throw an error if any of the variables are missing. not sure which are the values we can  handle gracefully and still have a running app while they are missing. i'm guessing ultimately having a network switcher will make sure we have all the stuff we need defined.